jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -2449,8 +2451,9 @@ public void login(String username, String password) throws ServletException
     public void logout() throws ServletException
     {
         if (_authentication instanceof Authentication.User)
-            ((Authentication.User)_authentication).logout();
-        _authentication=Authentication.UNAUTHENTICATED;
+            _authentication = ((Authentication.User)_authentication).logout(this);
```

Comment:
This could just be:
```java
if (_authentication instanceof Authentication.Logout)
￼             ((Authentication.Logout)_authentication).logout();
```

Reply:
Done.